spec/features/admin/users_spec.rb

```diff
@@ -247,7 +247,7 @@
 
       context 'on Punches tab' do
         let!(:monday_1_month_ago) { (DateTime.now - 4.week).monday }
-        let!(:datetime_first_monday) { user.office_holidays.include?({month: monday_1_month_ago.month, day: monday_1_month_ago.day}) ? monday_1_month_ago : monday_1_month_ago - 1.week }
+        let!(:datetime_first_monday) { user.office_holidays.include?({month: monday_1_month_ago.month, day: monday_1_month_ago.day}) ? monday_1_month_ago - 1.week : monday_1_month_ago }
```

Comment:
Wouldn't be better to use Timecop to freeze the time to some month we're sure we won't reach a holiday?